packages/react/src/components/Grid/Grid.js

```diff
@@ -8,30 +8,55 @@
 import { settings } from 'carbon-components';
 import cx from 'classnames';
 import PropTypes from 'prop-types';
-import React from 'react';
+import React, { useContext } from 'react';
+import { useFeatureFlag } from '../FeatureFlags';
 
 const { prefix } = settings;
 
+const GridContext = React.createContext({ hasGridParent: false });
```

Comment:
Might help to have this be:

```suggestion
const GridContext = React.createContext(false);
```

And then the value can be a primitive type / boolean. This can help since on every render of `Grid` the identity of the `value` will change when it's an object. That will then trigger re-renders for each child that uses `useContext(GridContext)` since the value changes

Reply:
Great point, fixed